src/webviews/webview-side/data-wrangler-html-renderer/dataWranglerHtmlRenderer.tsx

```diff
@@ -0,0 +1,156 @@
+            resolve(false);
+        }
+
+        // wait for a response from the host for additional confirmation
+        const listener = context.onDidReceiveMessage?.((e) => {
+            if (e.type === "shouldShowLaunchButtonResponse") {
+                listener?.dispose();
+                resolve(e.payload === true);
+            }
+        });
+
+        // post the query to the host for additional checking
+        postMessage({
+            outputId: id,
+            type: "shouldShowLaunchButton"
+        });
+    });
+}
+
+/**
+ * Overrides the default text/html renderer.
+ */
+export class DataWranglerHtmlRenderer extends React.PureComponent<IDataWranglerHtmlRendererProps, IDataWranglerHtmlRendererState> {
+    private containerRef = React.createRef<HTMLDivElement>();
+
+    override state = {
+        isDataFrame: false,
+        lostVariableName: undefined,
+        htmlToRender: undefined
+    };
+
+    override componentDidMount() {
+        const { outputItem, context, defaultRenderer } = this.props;
+
+        // when we first mount, just render using the default renderer
+        const element = this.containerRef.current;
+        if (element && defaultRenderer) {
+            defaultRenderer.renderOutputItem(outputItem, element);
+        }
+
+        // try to get the output text, if decoding failed then just default to empty string
+        let outputText = '';
+        try {
+            outputText = outputItem.text();
+        } catch (e) {}
+
+        // if the default renderer is somehow missing, we can still fall back to our own HTML rendering
+        if (!element || !defaultRenderer) {
+            this.setState({
+                htmlToRender: outputText
+            })
+        }
+
+        // if we ever get a message to hide the button, then we should just hide it
+        context.onDidReceiveMessage?.((e) => {
+            if (e.type === "variableWasLost") {
+                this.setState({
+                    isDataFrame: false,
+                    lostVariableName: e.payload
+                })
+            }
+        })
+
+        // fire off a check to see if the current output is a dataframe
+        void isPandasDataFrame(outputItem.id, outputText, context).then((isDataFrame) => {
+            if (isDataFrame) {
+                this.setState({
+                    isDataFrame
+                });
+            }
+        });
+    }
+
+    private launchInDataWrangler = () => {
+        const { outputItem, context } = this.props;
+        context.postMessage?.({
+            outputId: outputItem.id,
+            type: "launchDataWrangler"
+        });
+    };
+
+    override render() {
+        const { getLocalizedStrings } = this.props;
+        const { isDataFrame, lostVariableName, htmlToRender } = this.state;
+        const locStrings = getLocalizedStrings();
+        return (
+            <div>
```

Comment:
I'm a little worried about this. Maybe I'm misunderstanding though. After this renderer is registered, won't this div be around ALL html type output?

Reply:
that's correct, I can change it to not have the wrapper div around it if we're not rendering our button